File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 13/07/2026, 22:30:52
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map <int, int> hashmap;
        vector<int> solution;

        for (int i = 0; i < nums.size(); i++){
            int search = target - nums[i];
            if (hashmap.find(search) != hashmap.end()){
                solution.insert(solution.begin() , hashmap[search]);
                solution.insert(solution.begin() + 1 , i);
                break;
            }
            hashmap.insert({nums[i], i});
        }

        return solution;
    }
};

File: 1-TwoSum/1-TwoSum.py
# Last updated: 13/07/2026, 22:30:54
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = dict()
        for i in range(len(nums)):
            if (target - nums[i] in result):
                return [result[target-nums[i]] , i]
            result[nums[i]] = i
        return [-1,-1]